### ISSUE TITLE
Day cells are not always equal height with calendar height set to auto

### ISSUE THREAD

**Issue by lllopo** [fullcalendar/fullcalendar#5762]
### Reduced Test Case
[https://codesandbox.io/s/friendly-wiles-7efst?fontsize=14&hidenavigation=1&theme=dark](https://codesandbox.io/s/friendly-wiles-7efst?fontsize=14&hidenavigation=1&theme=dark)

### Bug Description
With height set to auto and number of events more than the minimum height of a calendar cell, the cells in GridView month calendar become unequal in height per row. The rows with events are higher then the rows without events.

### Screenshots

![calheight](https://user-images.githubusercontent.com/5485902/91557623-7e550780-e93d-11ea-92bc-008282a721e5.png)

**Comment by acerix** [comment]
You would need to make your calendar taller or show less events in each day so they can all fit.  If you want a different display, it's best if you can provide a mockup.

**Comment by lllopo** [comment]
@acerix I think you misunderstood the issue. It is not about if the events fit or not. They fit perfectly fine, just as I want them and as they were in FC 4. The problem is that the table rows (weeks) where there are no events are not the same height as those that have some. And this was not the case in FC 4. 
Simply said - all table rows should be the same height as the tallest one when, the height is auto, just the way it was in FC 4. Currently in FC 5 :
- Fixed height and dayMaxEventRows boolean -> rows are equal height. 
- Fixed height and dayMaxEventRows number that causes significant height -> rows are not equal. I guess you tried to avoid the scrolls as much as possible, but it looks really ugly (just as on the screenshot)
- Auto height, no matter the dayMaxEventRows value -> rows are not equal. Again - ugly.

**Comment by acerix** [comment]
Thanks for the additional info, I agree it looks better with equal row heights in general, and if it worked like that in v4 it seems like a bug.

**Comment by arshaw** [comment]
I have a longer response planned for this ticket @lllopo, however I'd first quickly like to point out that the v4 behavior ALSO results in rows that are uneven heights:

https://codepen.io/arshaw/pen/WNGOZGz?editors=0010

Each row has a min-height to prevent it from completely collapsing. The reason v5 looks worse than v4 is because this min-height is smaller, making the unevenness look more extreme:

https://codepen.io/arshaw/pen/KKgqXNB?editors=0010

**Comment by arshaw** [comment]
@lllopo, I agree with you that things look weird, but I don't think the rows should be unconditionally uniform when height is set to `auto`... that could result in some really large empty rows if just one row had lots of events. As hinted in my previous post, I believe a larger row min-height is the solution. [This new ticket](https://github.com/fullcalendar/fullcalendar/issues/6033) of mine explains how to accomplish that (spoiler: by using `aspectRatio` to determine a good cell min-height).

**Comment by lllopo** [comment]
@arshaw Well ... it depends in what perspective you look at it. You are completely right in general, but on other hand - look at it that way - I'm pretty sure no one is using the month view without a certain limit on the events shown. So, getting the weirdly big cells on many events would be a rare case actually. Even so - it would be users choice - either limit the events count or get the big cells. You can also do some sort of combination between equal height cells, max-height on them and then you will even get the auto event count limit back.

**Comment by nandorojo** [comment]
Is there a workaround to make the cells the same height?

**Comment by nandorojo** [comment]
If anyone can point me in the right direction, I could send a PR. I'd love to get this as an option.

**Comment by nandorojo** [comment]
For now, I implemented a hack to get this to work.

```ts
  useEffect(function setCalendarEventHeightHack() {
    // a bit unsafe: I'm just grabbing the table via a class name
    const calendarElement = document.getElementsByClassName(
      'fc-scrollgrid-sync-table'
    )[0]

    if (calendarElement.tagName == 'TABLE') {
      const trElements = calendarElement.getElementsByTagName('tr')

      for (let i = 0; i < trElements.length; i++) {
        const tr = trElements[i]

        tr.style.height = `${100 / trElements.length}%`
      }
    }
  })
```

**Comment by fnagel** [comment]
My client want equal row heights so I came up with this solution: 

```
.fc.fc-daygrid-day-frame {
    height: 125px;
 }
```

Is there any "correct" way to do this? I was a little confused there is no info about how to keep rows / cells the same height.

**Comment by slayermass** [comment]
> My client want equal row heights so I came up with this solution:
> 
> ```
> .fc.fc-daygrid-day-frame {
>     height: 125px;
>  }
> ```
> 
> Is there any "correct" way to do this? I was a little confused there is no info about how to keep rows / cells the same height.

This will cause scroll to appear sooner than default behaviour when having many events
![image](https://github.com/fullcalendar/fullcalendar/assets/7114520/8ad5b993-a446-474f-a3c9-a846abc07ff6)

**Comment by santiagopuentep** [comment]
Most top calendars today have the same height for each week. Changing the sizes makes the UI look very disorganized. It's also very weird how it keeps so much margin under the last event:

![2024-03-21_231243](https://github.com/fullcalendar/fullcalendar/assets/137173/98698686-743b-47c3-adc1-3285a155e114)

Does anyone know why the row grows so much every time an event is added to it? It looks as if there was the height of a full row under the last event in that slot. I'm trying to fix it with a hack but I don't understand why it's happening like this.

**Comment by Rodesc** [comment]
> Does anyone know why the row grows so much every time an event is added to it? It looks as if there was the height of a full row under the last event in that slot. I'm trying to fix it with a hack but I don't understand why it's happening like this.

Setting  ` height: 'auto'` in the options fixed it for me

**Comment by arshaw** [comment]
This has been fixed in [v7.0.0-beta.0](https://github.com/fullcalendar/fullcalendar-workspace/releases/tag/v7.0.0-beta.0) in accordance with what I wrote in #6033. When height is 'auto', and one of the rows has lots of events expanding its height, this does not affect the height of the other rows.

Updated repro:
https://codepen.io/arshaw/pen/ZEgQaqE?editors=0010

**Comment by Iuriy-Budnikov** [comment]
How to fix it in v6?
